pkgs/tools/networking/ntpsec/default.nix

```diff
@@ -0,0 +1,73 @@
+{ stdenv, lib, fetchurl, 
+  wafHook, pkgconfig, gnum4,
+  asciidoc, libxslt, docbook_xsl,
+  bison, openssl_1_1, libbsd,
+  coreutils, pythonPackages, pps-tools,
+  libcap ? null, libseccomp ? null,
+}:
+
+assert stdenv.isLinux -> libcap != null;
+assert stdenv.isLinux -> libseccomp != null;
+
+let
+  python = pythonPackages.python;
+
+  withSeccomp = stdenv.isLinux && (stdenv.isi686 || stdenv.isx86_64);
+in
+
+stdenv.mkDerivation rec {
+  name = "ntpsec-${version}";
+  version = "1.1.4";
+
+  src = fetchurl {
+    url = "https://ftp.ntpsec.org/pub/releases/ntpsec-${version}.tar.gz";
+    sha256 = "1vv5ckjlmaa9p2ri3fnqh804fxn0wl31jkkjn6v6hfw5z69vkwbn";
+  };
+
+  patches = [
+    ./skip-decodenetnum-tests.patch
+  ];
+
+  preBuild = ''
+    substituteInPlace ./ntpclients/ntpleapfetch \
+      --replace cat ${coreutils}/bin/cat \
+      --replace basename ${coreutils}/bin/basename \
+      --replace chmod ${coreutils}/bin/chmod
+  '';
+
+  wafConfigureFlags = [
+    "--sysconfdir=/etc"
+    "--localstatedir=/var"
+    "--refclock=all"
+    "--enable-leap-smear"
+    "--python=${python}/bin/python"
+  ]
+    ++ stdenv.lib.optional withSeccomp "--enable-seccomp";
+
+
+  nativeBuildInputs = [ pkgconfig wafHook gnum4
+    asciidoc docbook_xsl libxslt
+    python pythonPackages.wrapPython
+  ];
+
+  buildInputs = [ libbsd libcap bison openssl_1_1 python ]
+    ++ lib.optional withSeccomp libseccomp
+    ++ lib.optional stdenv.isLinux pps-tools;
```

Comment:
```suggestion
    ++ lib.optionals stdenv.isLinux [ libcap pps-tools ];
```

Reply:
This was copying the how it was done in ntp, the check is done in all-packages.nix. I am not sure why it was done this way, but indeed if I try doing it this way it does not get linked. I cannot explain why it does not get linked if configured this way.

Commit 458c8381e03b is what changes where this gets checked for ntp. However it's commit message is lacking as it implies only a version bump.

Reply:
hmm, I think it's a matter of personal taste, but I personally like to see all logic in the file that pertains to how to construct that package. 

If you were going to offer [many variations of a package](https://github.com/NixOS/nixpkgs/blob/6ea5cf34b211b9ddcdb8a30cead68ee461b65c13/pkgs/top-level/python-packages.nix#L5408), this would make sense to me. But in this case, it seems all of it could be put into the nix file itself.

I don't feel super strong about this, but I'm sure someone else might :)

Reply:
I actually like your way better, but I couldn't get it to work... Though I finally figured out that it was `optionals` and not just `optional` when putting an array after the check... Updated.

Reply:
`optionals` takes a list and returns either that list or the empty list.
`optional` takes an element and returns either that element wrapped in a list or the empty list.

Yes, I also think it's better to have all the logic in the package file, since it's not really a personal choice here but a platform choice that changes how to build the package.

Reply:
Yep, not sure where to find the documentation for lib.optional, but luckily it's [also just in a plain nix file](https://github.com/NixOS/nixpkgs/blob/b39f2efa182ea6e3d1be2a0e9e8a81771fdf4181/lib/lists.nix#L241).

Reply:
In the [nixpkgs manual](https://nixos.org/nixpkgs/manual/#function-library-lib.lists.optional)